src/MapBox.cc

```diff
@@ -57,13 +57,11 @@ void MapBox::_compute(Mapping const & map,
     detail::assertEqual(ubnd.size(), "ubnd.size()", nin, "Nin");
     if (maxOutCoord == 0) {
         maxOutCoord = nout;
-    } else {
-       if (maxOutCoord > nout) {
-           std::ostringstream os;
-           os << "maxOutCoord = " << maxOutCoord << " not in range [1, " << nout
-                << "], or 0 for all remaining";
-           throw std::invalid_argument(os.str());
-       }
+    } else if ((maxOutCoord < 0) || (maxOutCoord > nout)) {
```

Comment:
No, on the input argument. It's a struct-like class though the fields probably should be const.

That said, i think the macOutCoord field is not properly set if the input is 0. I'll fix that on a separate ticket. Also, this function probably should only take the map as an argument.